src/condor_io/condor_crypt_aesgcm.cpp

```diff
@@ -0,0 +1,522 @@
+/***************************************************************
+ *
+ * Copyright (C) 1990-2007, Condor Team, Computer Sciences Department,
+ * University of Wisconsin-Madison, WI.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+
+#include "condor_common.h"
+#include "condor_debug.h"
+#include "condor_crypt_aesgcm.h"
+
+#include <algorithm>
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+
+#define IV_SIZE 16
+#define MAC_SIZE 16
+
+unsigned char g_unset_iv[IV_SIZE] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+
+/*
+void Condor_Crypt_AESGCM::resetState()
+{
+	dprintf(D_NETWORK, "Condor_Crypt_AESGCM::resetState\n");
+	m_conn_crypto_state.reset(new ConnCryptoState());
+	while (!memcmp(m_conn_crypto_state->m_iv_enc.iv, g_unset_iv, IV_SIZE)) {
+		RAND_pseudo_bytes(m_conn_crypto_state->m_iv_enc.iv, IV_SIZE);
+	}
+	memset(m_conn_crypto_state->m_iv_dec.iv, '\0', IV_SIZE);
+}
+*/
+
+// this function is static
+void Condor_Crypt_AESGCM::initState(StreamCryptoState* stream_state)
+{
+	dprintf(D_NETWORK | D_BACKTRACE, "ZKM: ******: Condor_Crypt_AESGCM::initState for %p.\n", stream_state);
+	if (stream_state) {
+		// reset encrypt side
+		RAND_pseudo_bytes(stream_state->m_iv_enc.iv, IV_SIZE);
```

Comment:
There's an internal function for getting crypto bytes -- should we switch to that instead to keep things all in one place?